test/integration/test_pkg_signing.py

```diff
@@ -54,3 +54,90 @@ def verify(ref, artifacts_folder, signature_folder, files):
     assert "Verifying ref:  pkg/0.1" in c.out
     assert "VERIFYING  conanfile.py" not in c.out  # It doesn't re-verify previous contents
     assert "VERIFYING  conan_sources.tgz" in c.out
+
+
+def test_pkg_sign_canonical():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile1.py": GenConanfile("lib1ok", "0.1"),
+            "conanfile2.py": GenConanfile("lib2fail", "0.1"),  # This pkg fails when installed
+            "conanfile3.py": GenConanfile("lib3fail", "0.1")})  # This pkg should always fail
+    c.run("create conanfile1.py")
+    c.run("create conanfile2.py")
+    c.run("create conanfile3.py")
+    signer = textwrap.dedent(r"""
+        import os
+        from conan.errors import ConanException
+
+        def sign(ref, artifacts_folder, signature_folder, output, sign_tools):
+            output.info("Signing reference")
+            output.info(f"Signing folder: {artifacts_folder}")
+            if sign_tools.is_pkg_signed():
+                summary = sign_tools.load_summary()
+                if summary.get("provider") != "conan-client":
+                    return "Warn: Package already signed by another provider"
```

Comment:
This could be a warn or an error depending on the context or maybe there is some functionality still missing like `conan cache sign --force` to resign packages (user's plugin implementation). Also a `conan cache sign --clear` could be useful to "unsign" packages. WDYT?